### PR TITLE
New branch (rm)

### LINKED ISSUE #112: rm
reserving the rm command

### PR DESCRIPTION
**Pull request by nicklinck**
# Advanced HW 9 submission

Closes #112

Name: [Nick Linck]
Uniqname: [linckn]

### CODE REVIEW

File: _commands/basics/rm.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+---
+
+rm
+--
+
+`rm` is used to delete one or multiple files from your current directory. 
+
+~~~ bash
+$ rm file1.txt file2.txt
+
```
Comment (mterwill) [inline _commands/basics/rm.md]: There's trailing whitespace here that I'm personally not a fan of – remove this extra line for all of your code blocks? 

screen shot 2016-03-24 at 2 46 37 pm

File: _commands/basics/rm.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+file.txt
+
+~~~
+
+
+##### Break it down
+ 
+ * the '-v' command puts the rm command into verbose mode. This will print the name of the files that will be deleted.
+
+#### `echo -f`
```
Comment (mterwill) [inline _commands/basics/rm.md]: s/echo/rm

File: _commands/basics/rm.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+~~~ bash
+$ rm file1.txt file2.txt
+
+~~~
+
+<!--more-->
+
+### Useful Options / Examples
+
+#### `rm -d`
```
Comment (mterwill) [inline _commands/basics/rm.md]: I always do `rm -R`. I found [this](http://stackoverflow.com/questions/11976861/bash-whats-the-difference-between-rm-d-and-rm-r). Also in a `man rm` got 

           -d, --dir
              remove empty directories

which contradicts your explanation. Maybe @ppannuto can clear up some of the ambiguity?

File: _commands/basics/rm.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+---
+---
+
+rm
+--
+
+`rm` is used to delete one or multiple files from your current directory.
```
Comment (mterwill) [inline _commands/basics/rm.md]: Might be worth nothing that this does __not__ move to trash, but will delete the file permanently/immediately. Be careful!

File: _commands/basics/rm.md (hunk 3 above)
Comment (ppannuto) [inline _commands/basics/rm.md]: `-r` and `-R` (identical) both imply `-d`, from the OS X man page:

```
     -d          Attempt to remove directories as well as other types of
                 files.
     -R          Attempt to remove the file hierarchy rooted in each file
                 argument.  The -R option implies the -d option.  If the -i
                 option is specified, the user is prompted for confirmation
                 before each directory's contents are processed (as well as
                 before the attempt is made to remove the directory).  If
                 the user does not respond affirmatively, the file hierar-
                 chy rooted in that directory is skipped.

     -r          Equivalent to -R.
```

My habit is `-rf` as well, but not sure where that was learned or why to be honest.

**Comment by mterwill** [comment]
Looks like you committed some extra files in there, please clean that up!

**Comment by themoh** [comment]
We'll be closing all pull requests at midnight tonight, so please have this done by then.

File: _commands/basics/rm.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+~~~ bash
+$ rm -v file.txt
+file.txt
+~~~
+
+
+##### Break it down
+ 
+ * the '-v' command puts the rm command into verbose mode. This will print the name of the files that will be deleted.
```
Comment (mterwill) [inline _commands/basics/rm.md]: put `-v` inside \`backticks\`

File: _commands/basics/rm.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+#### `rm -f`
+
+~~~ bash
+$ rm -f file.txt
+~~~
+
+##### Break it down
+
+ * the '-f' flag allows the user who is using the rm command to remove the write-protected file without prompting the user.
```
Comment (mterwill) [inline _commands/basics/rm.md]: put `-f` inside \`backticks\`

File: _commands/basics/rm.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+~~~ bash
+$ rm -d directory1
+$ rm -d directory2
+rm: directory2: Directory not empty
+~~~
+
+
+##### Break it down
+ * The `-d` flag will remove directorys that are empty. In the example abpve, directory1 was empty and thus deleted while directory2 was not.
```
Comment (mterwill) [inline _commands/basics/rm.md]: Typo in "above"

**Comment by mterwill** [comment]
Still looks like you're trying to merge `reference.html`.

**Pull request merged**